### ISSUE TITLE
glz::obj + glz::json_t merging problems.

### ISSUE THREAD

**Issue by GJanos** [stephenberry/glaze#350]
When I try to read in a string containing a json, than later on try to merge that with a glz::obj, the serialization of the result is wrong. 

```c++
glz::json_t json;
    glz::read_json(json,"{\"key1\":42,\"key2\":\"hello world\",\"v\":[1,2,3],\"m\":{\"a\":1,\"b\":2,\"c\":3}}");
    glz::obj obj {"not", "important"};
    fmt::print("{}\n", glz::write_json(glz::merge{obj, json}));
    //{"not":"important","key1":42,"key2":"hello world","m":"a":1,"b":2,"c":3,"v":[1,2,3]}
    // -- map did not get properly serialized
```
All other types, except maps are serialized well.

**Comment by stephenberry** [comment]
Oh, this is doing a deeply nested merge. I guess this isn't desired behavior and you only want top level objects to be merged.

**Comment by stephenberry** [comment]
@GJanos
This problem should be fixed with #351. Let me know if you have further issues.

**Comment by GJanos** [comment]
@stephenberry Thank you, the problem is solved.
